sdk/provisioning/Azure.Provisioning/tests/Expressions/BicepFunctionTests/JoinTests.cs

```diff
@@ -0,0 +1,169 @@
+
+        // Act
+        var result = BicepFunction.Join(arrayValue, delimiter);
+
+        // Assert
+        TestHelpers.AssertExpression("join(myArray, '-')", result);
+    }
+
+    [Test]
+    public void Join_WithDelimiterVariable_ReturnsCorrectFormat()
+    {
+        // Arrange
+        var array = new BicepList<string> { "a", "b", "c" };
+        var delimiterVariable = new ProvisioningVariable("delimiter", typeof(string));
+        var delimiter = new BicepValue<string>(new IdentifierExpression("delimiter"));
+
+        // Act
+        var result = BicepFunction.Join(array, delimiter);
+
+        // Assert
+        TestHelpers.AssertExpression(
+            """
+            join([
+              'a'
+              'b'
+              'c'
+            ], delimiter)
+            """, result);
+    }
+
+    [Test]
+    public void Join_WithBothVariables_ReturnsCorrectFormat()
+    {
+        // Arrange
+        var itemsVariable = new ProvisioningVariable("items", typeof(string[]));
+        BicepList<string> arrayValue = itemsVariable;
+        var separatorVariable = new ProvisioningVariable("separator", typeof(string));
+        BicepValue<string> delimiter = separatorVariable;
+
+        // Act
+        var result = BicepFunction.Join(arrayValue, delimiter);
+
+        // Assert
+        TestHelpers.AssertExpression("join(items, separator)", result);
+    }
+
+    [Test]
+    public void Join_WithEmptyDelimiter_ReturnsCorrectFormat()
+    {
+        // Arrange
+        var array = new BicepList<string> { "Hello", "World" };
+        var delimiter = new BicepValue<string>("");
+
+        // Act
+        var result = BicepFunction.Join(array, delimiter);
+
+        // Assert
+        TestHelpers.AssertExpression(
+            """
+            join([
+              'Hello'
+              'World'
+            ], '')
+            """, result);
+    }
+
+    [Test]
+    public void Join_WithMultiCharDelimiter_ReturnsCorrectFormat()
+    {
+        // Arrange
+        var array = new BicepList<string> { "part1", "part2", "part3" };
+        var delimiter = new BicepValue<string>("::");
+
+        // Act
+        var result = BicepFunction.Join(array, delimiter);
+
+        // Assert
+        TestHelpers.AssertExpression(
+            """
+            join([
+              'part1'
+              'part2'
+              'part3'
+            ], '::')
+            """, result);
+    }
+
+    [Test]
+    public void Join_WithSingleElementArray_ReturnsCorrectFormat()
+    {
+        // Arrange
+        var array = new BicepList<string> { "onlyOne" };
+        var delimiter = new BicepValue<string>(",");
+
+        // Act
+        var result = BicepFunction.Join(array, delimiter);
+
+        // Assert
+        TestHelpers.AssertExpression(
+            """
+            join([
+              'onlyOne'
+            ], ',')
+            """, result);
+    }
+
+    [Test]
+    public void Join_WithResourceModelProperty_ReturnsCorrectFormat()
```

Comment:
let us add another test, which reference a property of a resource calling `ToBicepExpression` on it.